vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java

```diff
@@ -413,14 +455,36 @@ public void filterAndSelectItemNotOnFirstPage_setCurrentValue_valueCorrect() {
     public void setComponentRenderer_scrollDown_scrollUp_itemsRendered() {
         clickButton("component-renderer");
         beanBox.openPopup();
+
         scrollToItem(beanBox, 300);
+        waitUntilTextInContent("<h4>Person 300</h4>");
+
         scrollToItem(beanBox, 0);
+        waitUntilTextInContent("<h4>Person 0</h4>");
 
         assertComponentRendered("<h4>Person 0</h4>");
         assertComponentRendered("<h4>Person 4</h4>");
         assertComponentRendered("<h4>Person 9</h4>");
     }
 
+    @Test
+    public void scrollDown_scrollUp_selectionRetained() {
+        beanBox.sendKeys("Person 0");
+        waitUntilTextInContent("Person 0");
+        beanBox.sendKeys(Keys.ENTER);
+
+        beanBox.openPopup();
+        waitUntilTextInContent("Person 0");
+
+        scrollToItem(beanBox, 600);
+        waitUntilTextInContent("Person 600");
```

Comment:
This might turn into a false-positive in the future if we decide to increase the amount of items to keep in client-side from 500.

Would be more robust if we check that `getLoadedItems` (which basically returns `filteredItems`) doesn't include the first item anymore.